lms/models/course.py

```diff
@@ -0,0 +1,77 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy.orm.exc import NoResultFound
+
+from lms.db import BASE
+from lms.models.application_settings import ApplicationSettings
+
+
+class Course(BASE):
+    """An LTI course."""
+
+    __tablename__ = "course"
+
+    #: The authority_provided_id that uniquely identifies the course that these
+    #: settings belong to.
+    authority_provided_id = sa.Column(sa.UnicodeText(), primary_key=True)
+
+    #: The LTI consumer_key (oauth_consumer_key) of the application instance
+    #: that these course settings belong to.
+    consumer_key = sa.Column(
+        sa.String(),
+        sa.ForeignKey("application_instances.consumer_key", ondelete="cascade"),
+        primary_key=True,
+    )
+
+    _settings = sa.Column("settings", MutableDict.as_mutable(JSONB), nullable=False)
+
+    @property
+    def settings(self):
+        """
+        Return this course's settings.
+
+        :rtype: models.ApplicationSettings
+        """
+        return ApplicationSettings(self._settings)
+
+    @classmethod
+    def get(cls, db, authority_provided_id, consumer_key):
+        """Return the course matching `authority_provided_id` and `consumer_key` or None."""
+        try:
+            return cls._get(db, authority_provided_id, consumer_key)
+        except NoResultFound:
+            return None
```

Comment:
We only use the public version of `get` in the tests. I think we should collapse the two into one. I think we could do without the private one, and just check for return `None` instead of catching `NoResultFound` in `insert_if_not_exists` below.

Reply:
Yeah that'd work